### PR TITLE
refactor: consolidate actor scheduler configuration

### PR DESCRIPTION
**Pull request by romansmirnov**
## Description

This PR consolidates the actor scheduler bean declaration to avoid declaring the bean multiple times depending on which profile is used.

This step is a preparation step necessary to make the v2 endpoints (i.e., the endpoints defined in the module `gateway-rest`) available on every node running the REST API. That way, we can avoid implementing another bean declaration logic, for example, when the single application runs only with the profile `operate.`

## Related issues

related to #19076

### CODE REVIEW

**Review by houssain-barouni — APPROVED** [review]
LGTM

File: dist/src/main/java/io/camunda/commons/actor/ActorSchedulerConfiguration.java [hunk 1]
```diff
@@ -5,48 +5,55 @@
...
-
-    final var cpuThreads = cfg.getCpuThreadCount();
-    final var ioThreads = cfg.getIoThreadCount();
-    final var metricsEnabled = brokerCfg.getExperimental().getFeatures().isEnableActorMetrics();
+  public ActorScheduler scheduler() {
+    final var cpuThreads = properties.cpuThreads();
+    final var ioThreads = properties.ioThreads();
+    final var metricsEnabled = properties.metricsEnabled();
+    final var prefix = properties.prefix();
+    final var nodeId = properties.nodeId();
```
Comment (Zelldon) [inline dist/src/main/java/io/camunda/commons/actor/ActorSchedulerConfiguration.java:42]: 🔧 Maybe we can rename the properties field to the schedulerCfg

File: dist/src/main/java/io/camunda/zeebe/broker/BrokerModuleConfiguration.java [hunk 2]
```diff
@@ -42,13 +42,15 @@
...
     basePackages = {
+      "io.camunda.commons",
       "io.camunda.zeebe.broker",
       "io.camunda.zeebe.shared",
       "io.camunda.zeebe.gateway.rest",
       "io.camunda.authentication"
     })
 @ConfigurationPropertiesScan(
     basePackages = {
+      "io.camunda.commons",
```
Comment (Zelldon) [inline dist/src/main/java/io/camunda/zeebe/broker/BrokerModuleConfiguration.java:53]: 🤔 didn't you removed this in the other pr?

File: dist/src/main/java/io/camunda/zeebe/gateway/GatewayConfiguration.java [hunk 3]
```diff
@@ -46,6 +47,15 @@ public MultiTenancyCfg multiTenancyCfg() {
     return config.getMultiTenancy();
   }
 
+  @Bean
+  public SchedulerConfiguration schedulerConfiguration() {
+    final var cpuThreads = config.getThreads().getManagementThreads();
+    final var ioThreads = 0;
```
Comment (Zelldon) [inline dist/src/main/java/io/camunda/zeebe/gateway/GatewayConfiguration.java:53]:
```suggestion
    // We set ioThreads to zero as the Gateway isn't using any IO threads.
    final var ioThreads = 0;
```

Comment (Zelldon) [reply dist/src/main/java/io/camunda/zeebe/gateway/GatewayConfiguration.java:53]: Actually in general I think this is not really true, we could move some to ioThreads for requests etc as they io heavy, but right now we don't. AFAIK

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java [hunk 4]
```diff
@@ -344,9 +344,11 @@ private Broker createBroker(final int nodeId) {
             new BrokerClusterConfiguration().clusterConfig(brokerSpringConfig),
             Version.from(VersionUtil.getVersion()));
     final var scheduler =
-        new io.camunda.zeebe.broker.ActorSchedulerConfiguration(
-                brokerSpringConfig, actorClockConfiguration)
-            .scheduler(IdleStrategySupplier.ofDefault());
+        new io.camunda.commons.actor.ActorSchedulerConfiguration(
```
Comment (Zelldon) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java:347]: 🔧 I guess as we delete the other class we don't need to reference it via package name

**Review by ChrisKujawa — APPROVED** [review]
Thanks 👍🏼 I have smaller comments, but LGTM

File: dist/src/main/java/io/camunda/zeebe/broker/BrokerModuleConfiguration.java (hunk 2 above)
Comment (romansmirnov) [reply dist/src/main/java/io/camunda/zeebe/broker/BrokerModuleConfiguration.java:53]: Yep, the PRs are built upon each other. This intermediate step introduces this base package and removes it later. I tried to keep the PRs focused to better catch any issues.

Comment (Zelldon) [reply dist/src/main/java/io/camunda/zeebe/broker/BrokerModuleConfiguration.java:53]: I see 👍🏼

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java (hunk 4 above)
Comment (romansmirnov) [reply zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRule.java:347]: I think, it gets removed in one of the next PRs.

**Pull request merged**